joblib/test/test_func_inspect.py

```diff
@@ -72,6 +42,21 @@ def k(*args, **kwargs):
     pass
 
 
+@fixture(scope='module')
+def f2(tmpdir_factory):
+    # Create a Memory object to test decorated functions.
+    # We should be careful not to call the decorated functions, so that
+    # cache directories are not created in the temp dir.
```

Comment:
I think I got it eventually: we should only use f2 for testing its signature. If you call it you will pollute the temporary cache directory. The fact that it is a fixture makes it even more complicated to understand the comment I think ...

I may clean this up a bit after this PR is merged.